### PR TITLE
fix: new errors in conformance tests

### PR DESCRIPTION
**Pull request by bshaffer**

### CODE REVIEW

**Comment by matthewrobertson** [comment]
I think there are probably two issues that need to be correct in the cloud event conversion logic:

1. For Firebase RTDB you need to extract the source location from the legacy "domain" field as described [here](https://github.com/GoogleCloudPlatform/functions-framework-conformance/blob/master/docs/mapping.md#firebase-rtdb-events-tentative). [Here is the analogous PR in Ruby](https://github.com/GoogleCloudPlatform/functions-framework-ruby/pull/107/files)
2. For pubsub, you need to include the `messageId` and `publishTime` in the `message` object included in the CE data payload as described [here](https://github.com/GoogleCloudPlatform/functions-framework-conformance/blob/master/docs/mapping.md#cloud-pubsub-events). [Here is the analogous PR in Ruby](https://github.com/GoogleCloudPlatform/functions-framework-ruby/pull/105/files)

File: src/Context.php [hunk 1]
```diff
@@ -66,6 +69,11 @@ public function getResourceName(): ?string
         return $this->resource['name'] ?? null;
     }
 
+    public function getDomain(): ?string
+    {
+        return $this->domain ?? null;
```
Comment (dwsupplee) [inline src/Context.php:74]:
```suggestion
        return $this->domain;
```

File: src/LegacyEventMapper.php [hunk 2]
```diff
@@ -183,6 +192,20 @@ private function ceResourceAndSubject(string $ceService, string $resource): arra
                 $ret === 0 ? 'Resource regex did not match' : 'Failed while matching resource regex'
             );
         }
+        if ('firebasedatabase.googleapis.com' === $ceService) {
```
Comment (dwsupplee) [inline src/LegacyEventMapper.php:195]:
```suggestion
        if (self::FIREBASE_DB_CE_SERVICE === $ceService) {
```

**Review by dwsupplee — CHANGES_REQUESTED** [review]

File: src/LegacyEventMapper.php [hunk 3]
```diff
@@ -183,6 +192,20 @@ private function ceResourceAndSubject(string $ceService, string $resource): arra
...
+            if (null === $domain) {
+                return [null, null];
+            }
+            $location = 'us-central1';
+            if ($domain !== 'firebaseio.com') {
+                preg_match('#^([\w-]+)\.firebasedatabase\.app$#', $domain, $locationMatches);
```
Comment (matthewrobertson) [inline src/LegacyEventMapper.php:201]: We are actually more lenient than this in other languages as we aren't sure that `firebasedatabase.app` will be the only domain possible. You can basically match everything up to the first dot. 

[We weakened the regex in the Ruby FF after the original PR I pointed you to](https://github.com/GoogleCloudPlatform/functions-framework-ruby/pull/109/files)

File: src/Context.php (hunk 1 above)
Comment (bshaffer) [reply src/Context.php]: good point :P

File: src/LegacyEventMapper.php (hunk 2 above)
Comment (bshaffer) [reply src/LegacyEventMapper.php]: That's another good point!

File: src/LegacyEventMapper.php (hunk 3 above)
Comment (bshaffer) [reply src/LegacyEventMapper.php]: @matthewrobertson done in 42ae03dadb77227d5bfef17f1cf6999ef235a04d

**Review by matthewrobertson — APPROVED** [review]
LGTM

**Pull request merged**